framework/PostProcessors/EconomicRatio.py

```diff
@@ -0,0 +1,830 @@
+    # the insertion create another entry, this index should shift to the bigger side
+    indexL = utils.first(np.asarray(weightsCDF >= percent).nonzero())[0]
+    # This step returns the indices (list of index) of the array which is > than the percentile
+    indexH = utils.first(np.asarray(weightsCDF > percent).nonzero())
+    try:
+      # if the indices exists that means the desired percentile lies between two data points
+      # with index as indexL and indexH[0]. Calculate the midpoint of these two points
+      result = 0.5*(sortedWeightsAndPoints[indexL,1]+sortedWeightsAndPoints[indexH[0],1])
+    except IndexError:
+      result = sortedWeightsAndPoints[indexL,1]
+    return result
+
+  def _computeSortedWeightsAndPoints(self,arrayIn,pbWeight,percent):
+    """
+      Method to compute the sorted weights and points
+      @ In, arrayIn, list/numpy.array, the array of values from which the percentile needs to be estimated
+      @ In, pbWeight, list/numpy.array, the reliability weights that correspond to the values in 'array'
+      @ In, percent, float, the percentile that needs to be computed (between 0.01 and 1.0)
+      @ Out, sortedWeightsAndPoints, list/numpy.array, with [:,0] as the value of the probability density function at the bin, normalized, and [:,1] is the coresonding edge of the probability density function.
+      @ Out, indexL, index of the lower quantile
+    """
+
+    idxs                   = np.argsort(np.asarray(list(zip(pbWeight,arrayIn)))[:,1])
+    sortedWeightsAndPoints = np.asarray(list(zip(pbWeight[idxs],arrayIn[idxs])))
+    weightsCDF             = np.cumsum(sortedWeightsAndPoints[:,0])
+    indexL = utils.first(np.asarray(weightsCDF >= percent).nonzero())[0]
+    return sortedWeightsAndPoints, indexL
+
+  def __runLocal(self, inputData):
+    """
+      This method executes the postprocessor action. In this case, it computes all the requested statistical FOMs
+      @ In, inputData, tuple,  (inputDataset, pbWeights), tuple, the dataset of inputs and the corresponding
+        variable probability weight
+      @ Out, outputSet or outputDict, xarray.Dataset or dict, dataset or dictionary containing the results
+    """
+    inputDataset, pbWeights = inputData[0], inputData[1]
+    #storage dictionary for skipped metrics
+    self.skipped = {}
+    #construct a dict of required computations
+
+
+    needed = dict((metric,{'targets':set()}) for metric in self.scalarVals + self.tealVals)
+
+    needed.update(dict((metric,{'targets':set(),'threshold':{}}) for metric in ['sortinoRatio','gainLossRatio']))
+
+    needed.update(dict((metric,{'targets':set(),'threshold':[]}) for metric in ['valueAtRisk', 'expectedShortfall']))
+    for metric, params in self.toDo.items():
+      for entry in params:
+        needed[metric]['targets'].update(entry['targets'])
+        try:
+          needed[metric]['features'].update(entry['features'])
+        except KeyError:
+          pass
```

Comment:
how often can this happen?

Reply:
I am not sure about this one, if I remove this except, it is not going to set up the feature part...

Reply:
you're right, delete them